### PR TITLE
[JAVA_API] Wrapper for ov::PartialShape

### PR DESCRIPTION
**Pull request by rajatkrishna**
Details:

- Added JNI bindings for Partial Shape operations

### CODE REVIEW

File: modules/java_api/src/main/java/org/intel/openvino/PartialShape.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        super(addr);
+    }
+
+    /**
+     * Get the dimension at specified index of a partial shape.
+     *
+     * @param index The index of dimension.
+     * @return The particular dimension of partial shape.
+     */
+    public long get_dimension(int index) {
```
Comment (likholat) [inline modules/java_api/src/main/java/org/intel/openvino/PartialShape.java:19]: Please align with C++ API: this method supposed to return `Dimension` object (even according the comment above).
And we already have `get_length()` support for `Dimension`:
https://github.com/openvinotoolkit/openvino_contrib/blob/b0826c595d60616a7253382e3b04a4023f73b803/modules/java_api/src/main/java/org/intel/openvino/Dimension.java#L20
https://github.com/openvinotoolkit/openvino_contrib/blob/b0826c595d60616a7253382e3b04a4023f73b803/modules/java_api/src/main/cpp/dimension.cpp#L12

File: modules/java_api/src/main/cpp/partial_shape.cpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+#include "openvino_java.hpp"
+#include "jni_common.hpp"
+
+using namespace ov;
+
+JNIEXPORT jlong JNICALL Java_org_intel_openvino_PartialShape_GetDimension(JNIEnv *env, jobject obj, jlong addr, jint index) {
+    JNI_METHOD("GetDimension",
+        PartialShape* partial_shape = (PartialShape *)addr;
+        return (jlong) &(*partial_shape)[index];
```
Comment (allnes) [inline modules/java_api/src/main/cpp/partial_shape.cpp:15]:
```suggestion
        return (jlong) (partial_shape + index);
```

File: modules/java_api/src/test/java/org/intel/openvino/ModelTests.java [hunk 3]
```diff
@@ -44,6 +44,20 @@ public void testGetShape() {
...
     }
 
+    @Test
+    public void testGetPartialShape() {
+        ArrayList<Output> outputs = net.outputs();
+        int[] ref = new int[] {1, 10};
+
+        PartialShape partialShape = outputs.get(0).get_partial_shape();
+        for (int i = 0; i < ref.length; i++) {
+            Dimension dim = partialShape.get_dimension(i);
```
Comment (allnes) [inline modules/java_api/src/test/java/org/intel/openvino/ModelTests.java:54]: @likholat we aren't allocating memory in Demension class but I found method for delete memory
Could you provide more information about it?
https://github.com/openvinotoolkit/openvino_contrib/blob/b0826c595d60616a7253382e3b04a4023f73b803/modules/java_api/src/main/java/org/intel/openvino/Dimension.java#L28
https://github.com/openvinotoolkit/openvino_contrib/blob/b0826c595d60616a7253382e3b04a4023f73b803/modules/java_api/src/main/cpp/dimension.cpp#L24
https://github.com/openvinotoolkit/openvino_contrib/blob/b0826c595d60616a7253382e3b04a4023f73b803/modules/java_api/src/main/cpp/openvino_java.hpp#L110

**Comment by likholat** [comment]
@rajatkrishna please, merge the latest changes from master

File: modules/java_api/src/main/cpp/partial_shape.cpp (hunk 2 above)
Comment (rajatkrishna) [reply modules/java_api/src/main/cpp/partial_shape.cpp:15]: Offsetting by `index` does not access the `index`-th Dimension in PartialShape. Reverting this and using the overloaded subscript operator for now.

**Review by likholat — APPROVED** [review]

**Comment by likholat** [comment]
@ilya-lavrenov please merge this PR

**Comment by ilya-lavrenov** [comment]
build_jenkins

**Pull request merged**